crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -692,6 +781,52 @@ where
     }
 }
 
+pub struct MoveBundle<T> {
+    pub source: Entity,
+    pub target: Entity,
+    _bundle: PhantomData<T>,
+}
+
+impl<T> MoveBundle<T> {
+    /// Creates a new [`MoveBundle`] with given source and target IDs.
+    pub fn new(source: Entity, target: Entity) -> Self {
+        Self {
+            source,
+            target,
+            _bundle: PhantomData,
+        }
+    }
+}
+
+impl<T> Command for MoveBundle<T>
+where
+    T: Bundle + 'static,
+{
+    fn write(self, world: &mut World) {
+        let bundle_opt = if let Some(mut source) = world.get_entity_mut(self.source) {
+            source.remove_bundle::<T>()
+        } else {
+            panic!("Could not move a bundle (of type `{}`) from entity {:?} because it doesn't exist in this World.\n\
```

Comment:
There's a slight difference between the two:

> from entity

and

> to entity

Which means it might be a bit gnarly. e.g.

```rust
fn panic_fn<Ty>(verb: &'static str, preposition: &'static str, entity: Entity) {
        panic!("Could not {} a bundle (of type `{}`) {} entity {:?} because it doesn't exist in this World.\n\
                If this command was added to a newly spawned entity, ensure that you have not despawned that entity within the same stage.\n\
                This may have occurred due to system order ambiguity, or if the spawning system has multiple command buffers",
                verb, std::any::type_name::<Ty>(), preposition, entity);
}
```

Ye/nay?

Reply:
Yeah I like this.

Reply:
@alice-i-cecile @b-rief https://github.com/bevyengine/bevy/pull/3717/commits/72a905636c32c87d8d2ab528b621e5aff4de279b hows that look?

Reply:
I can do it for https://github.com/bevyengine/bevy/pull/3717/files#diff-ebaeb267b15dfe866ccce4f60af1f90f8fd44f31fa1b6ac7529f4b41f38b743bR576 too if you'd like?

Reply:
Yes please; let's keep this consistent.